File: scripts/generate-icons.js
#!/usr/bin/env node
/**
 * Generate mobile home screen icons for VERA
 * Creates PNG images with purple gradient orb design
 */

const fs = require('fs');
const path = require('path');
const { createCanvas } = require('canvas');

// Icon sizes to generate
const sizes = [
  { name: 'apple-touch-icon.png', size: 180, desc: 'Apple Touch Icon' },
  { name: 'android-chrome-192x192.png', size: 192, desc: 'Android Chrome 192x192' },
  { name: 'android-chrome-512x512.png', size: 512, desc: 'Android Chrome 512x512' }
];

// VERA's purple gradient colors
const gradientStart = '#667eea';  // Light purple
const gradientEnd = '#764ba2';    // Dark purple

function createGradientOrb(size) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // Create circular gradient from light to dark purple
  const gradient = ctx.createRadialGradient(
    size / 2, size / 2, 0,              // Inner circle center
    size / 2, size / 2, size / 2        // Outer circle center and radius
  );

  gradient.addColorStop(0, gradientStart);   // Light purple at center
  gradient.addColorStop(1, gradientEnd);     // Dark purple at edges

  // Fill the entire canvas with the gradient
  ctx.fillStyle = gradient;
  ctx.fillRect(0, 0, size, size);

  // Add subtle shimmer/shine effect
  const shineGradient = ctx.createRadialGradient(
    size * 0.3, size * 0.3, 0,
    size / 2, size / 2, size / 2
  );
  shineGradient.addColorStop(0, 'rgba(255, 255, 255, 0.15)');
  shineGradient.addColorStop(0.5, 'rgba(255, 255, 255, 0)');
  shineGradient.addColorStop(1, 'rgba(0, 0, 0, 0.1)');

  ctx.fillStyle = shineGradient;
  ctx.fillRect(0, 0, size, size);

  return canvas;
}

async function generateIcons() {
  const publicDir = path.join(__dirname, '..', 'public');

  // Ensure public directory exists
  if (!fs.existsSync(publicDir)) {
    fs.mkdirSync(publicDir, { recursive: true });
    console.log(`✅ Created public directory: ${publicDir}`);
  }

  console.log('\n🎨 Generating VERA mobile home screen icons...\n');

  for (const iconConfig of sizes) {
    try {
      const canvas = createGradientOrb(iconConfig.size);
      const buffer = canvas.toBuffer('image/png');
      const filePath = path.join(publicDir, iconConfig.name);

      fs.writeFileSync(filePath, buffer);
      console.log(`✅ Created: ${iconConfig.name} (${iconConfig.size}x${iconConfig.size}px)`);
      console.log(`   Location: ${filePath}`);
      console.log(`   Size: ${(buffer.length / 1024).toFixed(2)} KB\n`);
    } catch (error) {
      console.error(`❌ Error creating ${iconConfig.name}:`, error.message);
    }
  }

  console.log('✨ Icon generation complete!');
  console.log('\nIcons created in public/ directory:');
  sizes.forEach(icon => {
    console.log(`  - ${icon.name}`);
  });
}

// Run generation
generateIcons().catch(error => {
  console.error('Fatal error:', error);
  process.exit(1);
});

File: scripts/generate-icons.py
#!/usr/bin/env python3
"""
Generate mobile home screen icons for VERA
Creates PNG images with purple gradient orb design
"""

import os
import sys
from PIL import Image, ImageDraw

# VERA's purple gradient colors
GRADIENT_START = (102, 126, 234)   # #667eea - Light purple
GRADIENT_END = (118, 75, 162)      # #764ba2 - Dark purple

def create_gradient_orb(size):
    """Create a circular gradient orb image"""
    # Create image with RGBA to support transparency
    image = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    pixels = image.load()
    
    center_x = size / 2
    center_y = size / 2
    max_radius = size / 2
    
    # Draw gradient from center outward
    for x in range(size):
        for y in range(size):
            # Calculate distance from center
            dx = x - center_x
            dy = y - center_y
            distance = (dx * dx + dy * dy) ** 0.5
            
            # Calculate gradient ratio (0 at center, 1 at edges)
            if distance <= max_radius:
                ratio = distance / max_radius
                
                # Interpolate between start and end colors
                r = int(GRADIENT_START[0] * (1 - ratio) + GRADIENT_END[0] * ratio)
                g = int(GRADIENT_START[1] * (1 - ratio) + GRADIENT_END[1] * ratio)
                b = int(GRADIENT_START[2] * (1 - ratio) + GRADIENT_END[2] * ratio)
                
                # Add subtle shimmer effect
                shine_factor = 0.15 * (1 - ratio) if distance < max_radius * 0.3 else 0
                r = min(255, int(r + 255 * shine_factor))
                g = min(255, int(g + 255 * shine_factor))
                b = min(255, int(b + 255 * shine_factor))
                
                # Smooth alpha falloff at edges
                alpha = int(255 * (1 - (distance / max_radius) ** 2))
                pixels[x, y] = (r, g, b, alpha)
    
    return image

def generate_icons():
    """Generate all required icon sizes"""
    
    # Define icon configurations
    sizes = [
        ('apple-touch-icon.png', 180, 'Apple Touch Icon'),
        ('android-chrome-192x192.png', 192, 'Android Chrome 192x192'),
        ('android-chrome-512x512.png', 512, 'Android Chrome 512x512')
    ]
    
    # Get public directory
    script_dir = os.path.dirname(os.path.abspath(__file__))
    public_dir = os.path.join(os.path.dirname(script_dir), 'public')
    
    # Create public directory if it doesn't exist
    os.makedirs(public_dir, exist_ok=True)
    print(f'✅ Public directory ready: {public_dir}\n')
    
    print('🎨 Generating VERA mobile home screen icons...\n')
    
    for filename, size, description in sizes:
        try:
            # Create gradient orb
            image = create_gradient_orb(size)
            
            # Save as PNG
            filepath = os.path.join(public_dir, filename)
            image.save(filepath, 'PNG')
            
            # Get file size
            file_size = os.path.getsize(filepath)
            
            print(f'✅ Created: {filename} ({size}x{size}px)')
            print(f'   Description: {description}')
            print(f'   Location: {filepath}')
            print(f'   Size: {file_size / 1024:.2f} KB\n')
        
        except Exception as e:
            print(f'❌ Error creating {filename}: {str(e)}\n')
            return False
    
    print('✨ Icon generation complete!')
    print('\nIcons created in public/ directory:')
    for filename, size, _ in sizes:
        print(f'  ✓ {filename}')
    
    return True

if __name__ == '__main__':
    success = generate_icons()
    sys.exit(0 if success else 1)
